modular_skyrat/modules/autoairlock_override/code/engineering_override.dm

```diff
@@ -111,17 +126,38 @@
 	update_appearance()
 	return
 
-// Pulse to disable emergency access/engineering override and flash the red lights.
-/datum/wires/airlock/on_pulse(wire)
-	. = ..()
-	var/obj/machinery/door/airlock/airlock = holder
-	switch(wire)
-		if(WIRE_IDSCAN)
-			if(airlock.hasPower() && airlock.density)
-				airlock.do_animate("deny")
-				if(airlock.emergency)
-					airlock.emergency = FALSE
-					airlock.update_appearance()
-				if(airlock.engineering_override)
-					airlock.engineering_override = FALSE
-					airlock.update_appearance()
+// Manual override for when it's not orange alert.
+GLOBAL_VAR_INIT(force_eng_override, FALSE)
+/proc/toggle_eng_override()
+	if(!GLOB.force_eng_override)
+		GLOB.force_eng_override = TRUE
+		minor_announce("Engineering staff will have expanded access to areas of the station during the emergency.", "Engineering Emergency")
+		SEND_GLOBAL_SIGNAL(COMSIG_GLOB_FORCE_ENG_OVERRIDE, TRUE)
+		SSblackbox.record_feedback("nested tally", "keycard_auths", 1, list("engineer override access", "enabled"))
+		return
+	else
+		GLOB.force_eng_override = FALSE
+		minor_announce("Expanded engineering access has been revoked.", "Engineering Emergency")
+		var/level = SSsecurity_level.get_current_level_as_number()
+		SSblackbox.record_feedback("nested tally", "keycard_auths", 1, list("engineer override access", "disabled"))
+		if(level == SEC_LEVEL_ORANGE)
+			return
+		SEND_GLOBAL_SIGNAL(COMSIG_GLOB_FORCE_ENG_OVERRIDE, FALSE)
+		return
```

Comment:
This also just feels like it's a lot of repetition of code that could be done in more concise way.
```suggestion

	GLOB.force_eng_override = FALSE
	minor_announce("Expanded engineering access has been revoked.", "Engineering Emergency")
	var/level = SSsecurity_level.get_current_level_as_number()
	SSblackbox.record_feedback("nested tally", "keycard_auths", 1, list("engineer override access", "disabled"))
	if(level == SEC_LEVEL_ORANGE)
		return

	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_FORCE_ENG_OVERRIDE, FALSE)
```